### ISSUE TITLE
[chassis][202305] high cpu usage due to rsyslog_plugin process in swss/bgp dockers

### ISSUE THREAD

**Issue by anamehra** [sonic-net/sonic-buildimage#18335]
#### Description
High CPU usage was reported during sonic-mgmt run. While debugging, it was observed that the rsyslog_plugin process in swss and bgp dockers were consuming high cpu

```
Tasks: 1006 total,  14 running, 988 sleeping,   0 stopped,   4 zombie
%Cpu(s): 87.3 us, 12.3 sy,  0.0 ni,  0.2 id,  0.0 wa,  0.0 hi,  0.2 si,  0.0 st
MiB Mem :  31308.9 total,  11533.1 free,  14101.9 used,   5673.9 buff/cache
MiB Swap:      0.0 total,      0.0 free,      0.0 used.  16686.9 avail Mem 



PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND      
3289282 root      20   0   99672  10364   9164 R  41.4   0.0 106:18.25 rsyslog_plugin                                                                                                                                                                          
3288786 root      20   0   99672  10512   9312 R  41.1   0.0 105:21.97 rsyslog_plugin                                                                                                                                                                          
3289358 root      20   0   99672  10608   9404 R  40.8   0.0 106:15.01 rsyslog_plugin                                                                                                                                                                          
3290229 root      20   0   99672  10636   9432 R  38.2   0.0 105:55.20 rsyslog_plugin                                                                                                                                                                          
3289726 root      20   0   99672   9760   8772 R  34.5   0.0 101:24.32 rsyslog_plugin                                                                                                                                                                          
3287801 root      20   0   99672  10532   9332 R  31.9   0.0 101:48.93 rsyslog_plugin                                                                                                                                                                          
3288826 root      20   0   99672  10512   9316 R  31.6   0.0 101:00.06 rsyslog_plugin                                                                                                                                                                          
3281082 root      20   0  174376  80876  16212 R  30.9   0.3  86:24.12 python3                                                                                                                                                                                 
3287736 root      20   0 3228500 920356 194452 S  30.9   2.9  93:03.13 syncd                                                                                                                                                                                   
3289158 root      20   0   99672  10464   9264 R  30.6   0.0 101:33.52 rsyslog_plugin                                                                                                                                                                          
3288811 root      20   0 3301100 913572 195432 S  30.3   2.8  94:22.19 syncd                                                                                                                                                                                   
3290279 root      20   0   99672  10552   9352 R  30.3   0.0 101:00.70 rsyslog_plugin                                                                                                                                                                          
3287831 root      20   0 3301136 911652 195300 S  28.6   2.8  92:29.83 syncd                                                                                                                                                                                   
3288054 root      20   0 3223416 911184 193948 S  28.6   2.8  93:29.93 syncd                                                                                                                                                                                   
3290468 root      20   0   99672  10676   9476 R  28.3   0.0 101:31.45 rsyslog_plugin 
```

**Image version**
sonic-buildimage branch 202305 sha 0ed4e6c30be0da5a9f972652ad7d3d51ea64db5a

#### Steps to reproduce the issue:
1.
2.
3.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by anamehra** [comment]
hi @abdosi , @vperumal , for your viz.

**Comment by prabhataravind** [comment]
@abdosi please help take a look at this issue on chassis platforms.

**Comment by prabhataravind** [comment]
@qiluo-msft for viz. possibly related to eventd (structured syslog)

**Comment by zbud-msft** [comment]
Hi @anamehra is this a high priority issue / blocking issue?

**Comment by anamehra** [comment]
> Hi @anamehra is this a high priority issue / blocking issue?

We hit this issue during sonic-mgmt run on Chassis (multi-asic) on 202305 image. This causes high CPU affecting multiple test cases. Multiple test cases also failed due to the log analyzer reporting the ERR from the monit process for high CPU usage. snmp queries reported delays too. Wondering if this could also impact pizza boxes.

**Comment by abdosi** [comment]
@ysmanman @judyjoseph @arlakshm for viz. this should be applicable for master also.

@zbud-msft : do you have nay update on this ?

**Comment by arlakshm** [comment]
@kenneth-arista, @ysmanman, @saksarav-nokia, @mlok-nokia,  for viz...

**Comment by zbud-msft** [comment]
I will prioritize this issue ETA: 5/3

**Comment by vperumal** [comment]
@zbud-msft - we have noticed that one of the places the spike happens is during https://github.com/sonic-net/sonic-mgmt/blob/master/tests/test_pretest.py#L153. In this testcase, the test checks if syslog rate-limit-feature is supported and if supported it enables it and then disables it. After this testcase, we notice the cpu spike due to rsylog plugin process and this stays till either a config reload/reboot happens

**Comment by arlakshm** [comment]
@zbud-msft, please share the update on this issue.

**Comment by anamehra** [comment]
This is reproducible on master as well.
restart of rsyslogd itself is causing the issue.
```
docker exec -it bgp0 supervisorctl restart rsyslogd
```
I see the issue in swss and bgp dockers only.

**Comment by zbud-msft** [comment]
I will provide update by EOD

**Comment by zbud-msft** [comment]
Fixed by https://github.com/sonic-net/sonic-buildimage/pull/19035

**Comment by anamehra** [comment]
> Fixed by #19035

Thank you, @zbud-msft
